xe2: Update BMG fp16 row major M: 1-128 strategy

CaoZhongZ opened the pull request:
Initial tuning of fp16 data-type for M [1 - 128]. Benefits are significant. Strictly speaking we should tune range 1 ~ 200+ however no model uses range beyond 100 or so on BMG so far (small memory).

rjoursler commented:
I checked performance against a [range of shapes](https://github.com/user-attachments/files/21604121/test_batch.txt), it looks like there are a two regions with performance regressions associated with the new kernels. Can you check if these can be avoided?
<img width="621" height="570" alt="image" src="https://github.com/user-attachments/assets/40a180f0-a031-4373-9229-f898940ce7a5" />

Simonsays095 commented on src/gpu/intel/jit/gemm/selector/db/kernel.db:
```diff
@@ -1162,6 +1162,15 @@ auto _CATALOG_ = kcatalog::toArray({
 {{'G', "gemm", {"N", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 33, -1}, {-1, 48, -1}, {-1, 33, -1}, {-1, 48, -1}, {16, 16, 1}, "ABI"}, "at64x2 am32x2+m32@64 aB wg 4x1x4 ikr af k64 acb hi pt sr br sb64 sm sn sys nmk np", {16, (LoopType) 255, 128, {(LoopType) 209, (LoopType) 255, (LoopType) 2}, {262144, 262144, 16777216}, {262144, 262144, 16777216}, {16, 16, 64}, {4, 1, 4}, 1, (WGType) 1, 4101, 0, ...
 {{'G', "gemm", {"N", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 49, -1}, {-1, 128, -1}, {-1, 49, -1}, {-1, 128, -1}, {16, 16, 1}, "ABI"}, "at64 am32+m32@32 aB wg 2x2x2 ikr xaf st rr acb hi pt sr br sb64 sn sys nmk np", {16, (LoopType) 255, 128, {(LoopType) 209, (LoopType) 255, (LoopType) 2}, {262144, 524288, 16777216}, {262144, 524288, 16777216}, {16, 32, 64}, {2, 2, 2}, 1, (WGType) 1, 4357, 0, 8 ...
 {{'G', "gemm", {"N", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {1025, 129, 1}, {-1, 768, 3583}, {1025, 129, 1}, {-1, 768, 3583}, {16, 16, 1}, "ABIh"}, "at64x2+m64@8 am32x2+m32@8 aB wg 4x2 xaf st acb hi pt sr br sb64 sm sn grf256 sys np", {16, (LoopType) 255, 256, {(LoopType) 208, (LoopType) 255, (LoopType) 255}, {524288, 524288, 16777216}, {524288, 524288, 16777216}, {32, 32, 64}, {4, 2, 1}, 1, (WGTy ...
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 1, 8192}, {2048, 32, -1}, {-1, 1, 8192}, {2048, 32, -1}, {16, 16, 1}, "IAB"}, "at32+m32@32 am64+m64@64 aB wg 4x1x4 vav sr br sb64 bm0 bk0 sys xaf sm sn st np grf256 ikr", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {16, 16, 64}, {4, 1, 4}, 1, (WGType) 1, 261, 0 ...
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 1, 4096}, {2048, 32, 8191}, {-1, 1, 4096}, {2048, 32, 8191}, {16, 16, 1}, "IAB"}, "at32+m32@32 am64+m64@64 aB wg 4x1x2 vav sr br sb64 bm0 bk0 sys xaf sm sn st np grf256 ikr", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {16, 16, 64}, {4, 1, 2}, 1, (WGType) 1, 26 ...
```
Why are these two kernels fit using the W model, while the others are fit using the E model? With the restrictions it doesn't really matter, but that leads to my next question...

Have you tried measuring performance without accept/allow restrictions? If we can rely on the performance model instead, it will help with future optimization efforts.

CaoZhongZ replied:
No, we don't use performance model. And we use shapes to distinguish the strategy so W/E part is just fictions. How could we express this properly in strategy?

CaoZhongZ replied:
Okay, let me look into model E and see if it can capture the performance heuristics.

CaoZhongZ commented:
Haven't noticed that kv/afb was presented in strategy so might not work well with ikr for some shapes. Here is revised version, mitigated strange low efficiency M < 12 or so. Also improve relatively larger M.
<img width="1095" height="688" alt="image (7)" src="https://github.com/user-attachments/assets/9aa3633f-6244-405b-ac32-ec8902c7b7dc" />

CaoZhongZ commented:
My bad, update both main and tuning branch missed me some changes. There shouldn't be any ikr combine with kv/afb. Fixed, please try again. @Simonsays095

CaoZhongZ commented:
hmmm, still some regression. Let me clear them.

CaoZhongZ commented:
make test perf-gpu
set primitive=gpu:gemm

CaoZhongZ commented:
Hi @Simonsays095, we found out that the original hash id have some issues that'll lead to errors. I rebased this PR to main and updated my main branch. If I can't trigger the test, please help trigger it again.

Simonsays095 commented:
make test perf-gpu
set primitive=gpu:gemm

Yejing-Lai commented on src/gpu/intel/gemm/jit/selector/db/kernel.db:
```diff
@@ -1166,6 +1166,15 @@ auto _CATALOG_ = kcatalog::toArray({
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 1, 8192}, {2048, 32, -1}, {-1, 1, 8192}, {2048, 32, -1}, {16, 16, 1}, "IAB"}, "at32+m32@32 am64+m64@64 aB wg 4x1x4 vav sr br sb64 bm0 bk0 sys xaf sm sn st np grf256 ikr", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {16, 16, 64}, {4, 1, 4}, 1, (WGType) 1, 261, 0 ...
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 1, 4096}, {2048, 32, 8191}, {-1, 1, 4096}, {2048, 32, 8191}, {16, 16, 1}, "IAB"}, "at32+m32@32 am64+m64@64 aB wg 4x1x2 vav sr br sb64 bm0 bk0 sys xaf sm sn st np grf256 ikr", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {16, 16, 64}, {4, 1, 2}, 1, (WGType) 1, 26 ...
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 97, 4096}, {2048, 128, -1}, {-1, 97, 4096}, {2048, 128, -1}, {4, 8, 1}, "ABIpq"}, "at32x2+m64@32 am64+m64@64 aB wg 4x4 xaf st vav  pt sr br sb64 bk0 sm sn grf256 sys", {16, (LoopType) 255, 256, {(LoopType) 208, (LoopType) 255, (LoopType) 2}, {524288, 1048576, 16777216}, {524288, 1048576, 32}, {32, 32, 32}, {4, 4, 1}, 1, (WGType) 1, 1 ...
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {1280, 65, 8192}, {1280, 128, 8192}, {1280, 65, 8192}, {1280, 128, 8192}, {4, 8, 1}, "ABIpq"}, "at32x2+m64@64 am64+m64@64 aB wg 4x2x2 xaf vav sr br sb64 bk0 sm sn grf256 sys st pt ikr", {16, (LoopType) 255, 256, {(LoopType) 208, (LoopType) 255, (LoopType) 2}, {524288, 1048576, 16777216}, {524288, 1048576, 32}, {32, 32, 32}, {4, 2, 2}, 1,  ...
```
It is expected. In our test case(qwen-32b, qwen-14b, and llama-70b), we found that only N=1280 and K=8192 have worse performance.

CaoZhongZ commented on src/gpu/intel/gemm/jit/selector/db/kernel.db:
```diff
@@ -1166,6 +1166,15 @@ auto _CATALOG_ = kcatalog::toArray({
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {1280, 65, 8192}, {1280, 128, 8192}, {1280, 65, 8192}, {1280, 128, 8192}, {4, 8, 1}, "ABIpq"}, "at32x2+m64@64 am64+m64@64 aB wg 4x2x2 xaf vav sr br sb64 bk0 sm sn grf256 sys st pt ikr", {16, (LoopType) 255, 256, {(LoopType) 208, (LoopType) 255, (LoopType) 2}, {524288, 1048576, 16777216}, {524288, 1048576, 32}, {32, 32, 32}, {4, 2, 2}, 1,  ...
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 1, -1}, {-1, 32, 1280}, {-1, 1, -1}, {-1, 32, 1280}, {4, 8, 1}, "ABI"}, "at32+m32@32 am64+m64@64 aB wg 4x2 xaf st vav pt sr br sb64 bk0 sm sn grf256 sys", {16, (LoopType) 255, 256, {(LoopType) 208, (LoopType) 255, (LoopType) 2}, {262144, 262144, 16777216}, {262144, 262144, 32}, {16, 16, 32}, {4, 2, 1}, 1, (WGType) 1, 445, 0, 8192, {4 ...
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 1, -1}, {-1, 32, -1}, {-1, 1, -1}, {-1, 32, -1}, {4, 8, 1}, "ABI"}, "at32+m32@64 am64+m64@64 aB wg 2x2x8 kr xaf st vav hi pt sr br sb64 bk0 sm sn grf256 sys", {16, (LoopType) 255, 256, {(LoopType) 208, (LoopType) 255, (LoopType) 2}, {262144, 262144, 16777216}, {262144, 262144, 32}, {16, 16, 32}, {2, 2, 8}, 1, (WGType) 1, 261, 0, 8192 ...
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, -33, -1}, {-1, 64, -1}, {-1, 33, -1}, {-1, 64, -1}, {4, 8, 1}, "ABI"}, "at32x2+m64@64 am64+m64@64 aB wg 4x4 xaf vav sr br sb64 bk0 sm sn grf256 sys st pt", {16, (LoopType) 255, 256, {(LoopType) 208, (LoopType) 255, (LoopType) 2}, {524288, 524288, 16777216}, {524288, 524288, 32}, {16, 16, 32}, {4, 4, 1}, 1, (WGType) 1, 1, 0, 32768, {4 ...
```
Updated.

CaoZhongZ commented:
Hi @Simonsays095 please help merge if there no further problem. Thanks!

skazakov1 commented:
my status: was trying to collect performance on BMG with mode=P suggested by Simon, met some issue with 1perf on comparison step. It seems we fixed it, but now I see TIMEOUT. Will try again. Also will run our regular perf CI here.

skazakov1 commented:
make test perf-gpu
set primitive=gpu:gemm

skazakov1 commented:
the latest perf CI (https://ecmd.jf.intel.com/commander/link/jobDetails/jobs/f09ece86-c600-f183-b36d-a4bf010d0e2d) showed the following models regression on BMG:
| p3.text-recognition-0012_tf.blocked.inf.fp16.ov.b32                                                 |    1.614706 |    1.764706 |  0.92 |
| - inner_product                                                                                     |    0.804086 |    0.954086 |  0.84 |
| p0.phi-2.FP16-4BIT.inf.fp16.ov.baseline_mb1024                                                      |   63.156544 |   69.813216 |  0.90 |
| - matmul                                                                                            |   63.156544 |   69.813216 |  0.90 |
| p3.text-recognition-0012_tf.blocked.inf.int8.ov.b32                                                 |    1.221060 |    1.371060 |  0.89 |
| - inner_product                                                                                     |    0.804086 |    0.954086 |  0.84 |
| p1.crnn-tf.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-tf                              |    1.985730 |    2.243630 |  0.89 |
| - inner_product                                                                                     |    1.684725 |    1.942625 |  0.87 |
| p1.crnn-tf.ir.inf.int8.ov.device-xeon-platinum-8352Y-atsm-framework-tf                              |    1.926890 |    2.184790 |  0.88 |
| - inner_product                                                                                     |    1.684725 |    1.942625 |  0.87 |
| p1.text-recognition-0012.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-tf                |    4.321445 |    4.943000 |  0.87 |
| - inner_product                                                                                     |    4.019910 |    4.641465 |  0.87 |
| p1.text-recognition-0012.ir.inf.int8.ov.device-xeon-platinum-8352Y-atsm-framework-tf                |    4.258950 |    4.883630 |  0.87 |
| - inner_product                                                                                     |    4.016785 |    4.641465 |  0.87 |
| p0.Llama-3.2-1B-Instruct.FP16.inf.fp16.pt.mb32                                                      | 1061.224015 | 1226.290401 |  0.87 |
| - matmul                                                                                            | 1061.224015 | 1226.290401 |  0.87 |
| p0.minicpm4-0.5b.FP16.inf.fp16.pt.llm_32_prompt                                                     |    2.193492 |    2.761846 |  0.79 |
| - matmul                                                                                            |    2.192760 |    2.761114 |  0.79 |
| p0.qwen2.5-1.5b-instruct.FP16.inf.fp16.pt.llm_32_prompt                                             |    7.727350 |   10.078600 |  0.77 |
| - matmul                                                                                            |    7.727350 |   10.078600 |  0.77 |
| p1.rnnt-prediction.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx                    |    0.034370 |    0.045830 |  0.75 |
| - inner_product                                                                                     |    0.034370 |    0.045830 |  0.75 |
| p0.gemma-3-1b-it.FP16.inf.fp16.pt.llm_32_prompt                                                     |    3.953300 |    5.369772 |  0.74 |
| - matmul                                                                                            |    3.952568 |    5.369040 |  0.74 |
| p0.ebgan.default.inf.fp16.tf.intel22                                                                |    0.463710 |    0.635495 |  0.73 |
| - matmul                                                                                            |    0.173519 |    0.345304 |  0.50 |
| p1.ebgan.ir.inf.fp32.ov.device-xeon-platinum-8352Y-atsm-framework-tf                                |    0.138530 |    0.190095 |  0.73 |
| - inner_product                                                                                     |    0.051035 |    0.102600 |  0.50 |
| p1.ebgan.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-tf                                |    0.136445 |    0.188010 |  0.73 |
| - inner_product                                                                                     |    0.051035 |    0.102600 |  0.50 |
| p3.tedlium_dnn4_smbr_nnet.blocked.inf.fp16.ov.b32                                                   |    0.070414 |    0.098224 |  0.72 |
| - inner_product                                                                                     |    0.070414 |    0.098224 |  0.72 | 

not so bed taking account great improvement:
p1.mozilla-deepspeech-0_8_2.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-tf             |   17.529110 |    6.714030 |  2.61 |
| - inner_product                                                                                     |   17.529110 |    6.714030 |  2.61 |
| p0.Phi-3.5-mini-instruct.FP16.inf.fp16.pt.mb1                                                       | 3108.336564 | 1298.057520 |  2.39 |
| - matmul                                                                                            | 3108.336564 | 1298.057520 |  2.39 |
| p1.tedlium-dnn4-smbr.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-kaldi                 |    1.110400 |    0.475505 |  2.34 |
| - inner_product                                                                                     |    1.110400 |    0.475505 |  2.34 |
| p0.chatglm2-6b.inf.fp16.ov.mb1                                                                      |  354.712491 |  152.288312 |  2.33 |
| - inner_product                                                                                     |  353.414747 |  150.970212 |  2.34 |
| - matmul                                                                                            |    1.297744 |    1.318100 |  0.98 |
| p0.synth_matmul_LLM.2nd.f16_f16_f16.none.inf.fp16.ov.no_postops                                     |   41.425706 |   19.589773 |  2.11 |
| - matmul                                                                                            |   41.425706 |   19.589773 |  2.11 |
| p0.synth_matmul_LLM.2nd.bf16_bf16_bf16.none.inf.bf16.ov.no_postops                                  |   41.419455 |   19.593415 |  2.11 |
| - matmul                                                                                            |   41.419455 |   19.593415 |  2.11 |
| p0.Llama-3.2-1B-Instruct.FP16.inf.fp16.pt.mb1                                                       | 1086.619777 |  528.986017 |  2.05 |
| - matmul                                                                                            | 1086.619777 |  528.986017 |  2.05 |
| p0.mozilla-deepspeech-0.8.2.default.inf.fp16.tf.public151                                           |   66.296158 |   32.437870 |  2.04 |
| - matmul                                                                                            |   66.296158 |   32.437870 |  2.04 |
| p0.mozilla-deepspeech-0.8.2.default.inf.int8.tf.public54                                            |   64.690100 |   32.023580 |  2.02 |
| - matmul                                                                                            |   64.690100 |   32.023580 |  2.02 |
| p1.alexnet.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx                            |    2.158310 |    1.119245 |  1.93 |
| - inner_product                                                                                     |    1.769265 |    0.730200 |  2.42 |
| p0.alexnet.default.inf.fp16.onnx.public56                                                           |    7.356480 |    3.823659 |  1.92 |
| - matmul                                                                                            |    6.015501 |    2.482680 |  2.42 |
| p1.alexnet.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-caffe                           |    2.207265 |    1.168200 |  1.89 |
| - inner_product                                                                                     |    1.769265 |    0.730200 |  2.42 |
| p0.qwen2-vl-7b-instruct.FP16.inf.fp16.ov.llm_32_prompt                                              |  158.623399 |   85.792087 |  1.85 |
| - matmul                                                                                            |  158.586734 |   85.755422 |  1.85 |
| p0.mixtral-8x7b-v0.1.FP16.inf.fp16.pt.llm_32_prompt                                                 |   82.186321 |   44.876498 |  1.83 |
| - matmul                                                                                            |   82.185589 |   44.875766 |  1.83 |
| p0.flan-t5-xxl.FP16.inf.fp16.pt.llm_32_prompt                                                       |   86.290576 |   48.399804 |  1.78 |
| - matmul                                                                                            |   86.290576 |   48.399804 |  1.78 |
| p0.qwen2.5-vl-3b-instruct.FP16.inf.fp16.ov.llm_32_prompt                                            |   72.073543 |   41.033869 |  1.76 |
| - matmul                                                                                            |   72.036878 |   40.997204 |  1.76 |
| p0.starcoder2-15b.FP16.inf.fp16.pt.llm_32_prompt                                                    |  159.858740 |   95.041850 |  1.68 |
| - matmul                                                                                            |  159.858740 |   95.041850 |  1.68 |
| p0.Phi-3.5-mini-instruct.FP16.inf.fp16.pt.mb8                                                       | 2254.950728 | 1365.517024 |  1.65 |
| - matmul                                                                                            | 2254.950728 | 1365.517024 |  1.65 |
| p0.phi-4-mini-instruct.FP16.inf.fp16.pt.llm_32_prompt                                               |   36.275260 |   22.150908 |  1.64 |
| - matmul                                                                                            |   36.275260 |   22.150908 |  1.64 |
| p0.chatglm3-6b.FP16.inf.fp16.pt.llm_32_prompt                                                       |   54.128668 |   33.226132 |  1.63 |
| - matmul                                                                                            |   54.128668 |   33.226132 |  1.63 |
| p0.gemma-2-9b-it.FP16.inf.fp16.pt.llm_32_prompt                                                     |   85.516972 |   53.107612 |  1.61 |
| - matmul                                                                                            |   85.516240 |   53.106880 |  1.61 |
| p0.minicpm4-8b.FP16.inf.fp16.pt.llm_32_prompt                                                       |   72.511192 |   45.220524 |  1.60 |
| - matmul                                                                                            |   72.510460 |   45.219792 |  1.60 |
| p0.gpt-j-6b.FP16.inf.fp16.pt.llm_32_prompt                                                          |   53.015340 |   33.157284 |  1.60 |
| - matmul                                                                                            |   53.015340 |   33.157284 |  1.60 |
| p0.phi-4.FP16.inf.fp16.pt.llm_32_prompt                                                             |  124.705100 |   78.099060 |  1.60 |
| - matmul                                                                                            |  124.705100 |   78.099060 |  1.60 |
| p0.qwen3-8b.FP16.inf.fp16.pt.llm_32_prompt                                                          |   69.037068 |   43.281228 |  1.60 |
| - matmul                                                                                            |   69.037068 |   43.281228 |  1.60 |
| p0.internvl2-4b.FP16.inf.fp16.ov.llm_32_prompt                                                      |  115.425248 |   72.847869 |  1.58 |
| - matmul                                                                                            |  115.334480 |   72.757101 |  1.59 |
| p0.llama-3.1-8b-instruct.FP16.inf.fp16.pt.llm_32_prompt                                             |   67.504092 |   42.610100 |  1.58 |
| - matmul                                                                                            |   67.503360 |   42.609368 |  1.58 |
| p0.baichuan2-13b-chat.FP16.inf.fp16.pt.llm_32_prompt                                                |  118.792520 |   75.043400 |  1.58 |
| - matmul                                                                                            |  118.792520 |   75.043400 |  1.58 |
| p0.gemma-7b-it.FP16.inf.fp16.pt.llm_32_prompt                                                       |   85.709292 |   54.215600 |  1.58 |
| - matmul                                                                                            |   85.708560 |   54.214868 |  1.58 |
| p0.qwen2.5-7b-instruct.FP16.inf.fp16.pt.llm_32_prompt                                               |   65.367688 |   41.578612 |  1.57 |
| - matmul                                                                                            |   65.367688 |   41.578612 |  1.57 |
| p0.qwen1.5-14b-chat.FP16.inf.fp16.pt.llm_32_prompt                                                  |  117.062880 |   74.548460 |  1.57 |
| - matmul                                                                                            |  117.062880 |   74.548460 |  1.57 |
| p0.llama-2-13b-chat-hf.FP16.inf.fp16.pt.llm_32_prompt                                               |  111.714012 |   71.450202 |  1.56 |
| - matmul                                                                                            |  111.713280 |   71.449470 |  1.56 |
| p0.phi-2.FP16.inf.fp16.pt.llm_32_prompt                                                             |   22.619668 |   14.527488 |  1.56 |
| - matmul                                                                                            |   22.619668 |   14.527488 |  1.56 |
| p0.llama-3.2-1b-instruct.FP16.inf.fp16.pt.llm_32_prompt                                             |   12.233992 |    7.909190 |  1.55 |
| - matmul                                                                                            |   12.233260 |    7.908458 |  1.55 |
| p0.phi-3-medium-4k-instruct.FP16.inf.fp16.pt.llm_32_prompt                                          |  129.808508 |   84.108498 |  1.54 |
| - matmul                                                                                            |  129.808020 |   84.108010 |  1.54 |
| p0.mistral-7b-instruct-v0.3.FP16.inf.fp16.pt.llm_32_prompt                                          |   62.595336 |   40.687180 |  1.54 |
| - matmul                                                                                            |   62.594604 |   40.686448 |  1.54 |
| p0.mistral-7b-v0.1.FP16.inf.fp16.pt.llm_32_prompt                                                   |   62.553878 |   40.670096 |  1.54 |
| - matmul                                                                                            |   62.553146 |   40.669364 |  1.54 |
| p0.llama-2-7b-chat-hf.FP16.inf.fp16.pt.llm_32_prompt                                                |   57.730582 |   37.646736 |  1.53 |
| - matmul                                                                                            |   57.729850 |   37.646004 |  1.53 |
| p0.phi-3-mini-4k-instruct.FP16.inf.fp16.pt.llm_32_prompt                                            |   32.838052 |   21.679104 |  1.51 |
| - matmul                                                                                            |   32.837564 |   21.678616 |  1.51 |
| p0.qwen2.5-coder-3b-instruct.FP16.inf.fp16.pt.llm_32_prompt                                         |   28.646916 |   19.031646 |  1.51 |
| - matmul                                                                                            |   28.646916 |   19.031646 |  1.51 |
| p0.llama-3.2-3b-instruct.FP16.inf.fp16.pt.llm_32_prompt                                             |   27.951784 |   18.836384 |  1.48 |
| - matmul                                                                                            |   27.951052 |   18.835652 |  1.48 |
| p0.minicpm3-4b.FP16.inf.fp16.pt.llm_32_prompt                                                       |   34.503409 |   23.640899 |  1.46 |
| - matmul                                                                                            |   34.502677 |   23.640167 |  1.46 |
| p0.falcon-7b-instruct.FP16.inf.fp16.pt.llm_32_prompt                                                |   63.310956 |   43.477014 |  1.46 |
| - matmul                                                                                            |   63.310956 |   43.477014 |  1.46 |
| p0.llama-3-8b-instruct.FP16.inf.fp16.pt.llm_32_prompt                                               |   70.754140 |   49.050132 |  1.44 |
| - matmul                                                                                            |   70.753408 |   49.049400 |  1.44 |
| p0.qwen2.5-vl-3b-instruct.FP16.inf.fp16.ov.llm_1024_prompt                                          |  146.957625 |  112.715658 |  1.30 |
| - matmul                                                                                            |  146.920960 |  112.678993 |  1.30 |
| p0.phi-4-multimodal-instruct.FP16.inf.fp16.ov.llm_32_prompt                                         |  183.721877 |  141.425694 |  1.30 |
| - matmul                                                                                            |  183.202712 |  140.906529 |  1.30 |
| p0.phi-3.5-vision-instruct.FP16.inf.fp16.ov.llm_32_prompt                                           |  188.871261 |  146.093343 |  1.29 |
| - matmul                                                                                            |  188.686468 |  145.908550 |  1.29 |
| p0.internvl2-4b.FP16.inf.fp16.ov.llm_1024_prompt                                                    |  196.141854 |  153.350263 |  1.28 |
| - matmul                                                                                            |  196.051086 |  153.259495 |  1.28 |
| p0.qwen2-vl-7b-instruct.FP16.inf.fp16.ov.llm_1024_prompt                                            |  363.865848 |  285.846830 |  1.27 |
| - matmul                                                                                            |  363.829183 |  285.810165 |  1.27 |
| p0.phi-3.5-vision-instruct.FP16.inf.fp16.ov.llm_1024_prompt                                         |  209.194061 |  166.402247 |  1.26 |
| - matmul                                                                                            |  209.009268 |  166.217454 |  1.26 |
| p0.phi-4-multimodal-instruct.FP16.inf.fp16.ov.llm_1024_prompt                                       |  213.428662 |  171.633433 |  1.24 |
| - matmul                                                                                            |  212.909497 |  171.114269 |  1.24 |
| p0.qwen2.5-3b-instruct-gptq.GPTQ_INT4-FP16.inf.int4.pt.llm_32_prompt                                |   12.623172 |   10.170462 |  1.24 |
| - matmul                                                                                            |   12.623172 |   10.170462 |  1.24 |
| p0.Phi-3.5-mini-instruct.FP16.inf.fp16.pt.mb16                                                      | 1958.587064 | 1632.682430 |  1.20 |
| - matmul                                                                                            | 1958.587064 | 1632.682430 |  1.20 |
| p0.synth_matmul_LLM.1st_thin.bf16_bf16_bf16.none.inf.bf16.ov.no_postops                             |   22.799349 |   19.159251 |  1.19 |
| - matmul                                                                                            |   22.799349 |   19.159251 |  1.19 |
| p0.synth_matmul_LLM.1st_thin.f16_f16_f16.none.inf.fp16.ov.no_postops                                |   22.653005 |   19.053001 |  1.19 |
| - matmul                                                                                            |   22.653005 |   19.053001 |  1.19 |
| p0.phi-4-mini-instruct.FP16.inf.fp16.pt.llm_1024_prompt                                             |  107.592092 |   90.540956 |  1.19 |
| - matmul                                                                                            |  107.592092 |   90.540956 |  1.19 |
| p0.DeepSeek-R1-Distill-Qwen-14B.BF16-4BIT.inf.bf16.pt.gs64_zp-asym_mb8                              | 2931.206665 | 2498.721585 |  1.17 |
| - matmul                                                                                            | 2931.206665 | 2498.721585 |  1.17 |
| p0.phi-2.FP16.inf.fp16.pt.llm_1024_prompt                                                           |   78.409684 |   67.270848 |  1.17 |
| - matmul                                                                                            |   78.409684 |   67.270848 |  1.17 |
| p0.stable-diffusion-v1-4-text-encoder.default.inf.fp16.onnx.public211                               |   11.325876 |    9.732228 |  1.16 |
| - matmul                                                                                            |   11.325876 |    9.732228 |  1.16 |
| p0.llama3.1-8b.finetune.tr.bf16.pt                                                                  |  126.966937 |  110.053585 |  1.15 |
| - matmul                                                                                            |  126.966937 |  110.053585 |  1.15 |
| p0.qwen2-7b-instruct.FP16.inf.fp16.pt.llm_1024_prompt                                               |  209.824224 |  182.091824 |  1.15 |
| - matmul                                                                                            |  209.824224 |  182.091824 |  1.15 |
| p0.phi-4.FP16.inf.fp16.pt.llm_1024_prompt                                                           |  406.455260 |  353.616020 |  1.15 |
| - matmul                                                                                            |  406.455260 |  353.616020 |  1.15 |
| p0.minicpm4-8b.FP16.inf.fp16.pt.llm_1024_prompt                                                     |  237.357784 |  206.870508 |  1.15 |
| - matmul                                                                                            |  237.357052 |  206.869776 |  1.15 |
| p0.phi-3-mini-4k-instruct.FP16.inf.fp16.pt.llm_1024_prompt                                          |  115.414884 |  100.649152 |  1.15 |
| - matmul                                                                                            |  115.414396 |  100.648664 |  1.15 |
| p0.Llama-3.2-1B-Instruct.FP16.inf.fp16.pt.mb16                                                      |  910.211909 |  795.385613 |  1.14 |
| - matmul                                                                                            |  910.211909 |  795.385613 |  1.14 |
| p0.DeepSeek-R1-Distill-Qwen-14B.BF16-4BIT.inf.bf16.pt.gs64_zp-asym_mb4                              | 3099.117628 | 2710.885166 |  1.14 |
| - matmul                                                                                            | 3099.117628 | 2710.885166 |  1.14 |
| p0.sbert-base-mean-tokens.default.inf.fp16.onnx.public185                                           |   14.831820 |   13.003980 |  1.14 |
| - matmul                                                                                            |   14.831820 |   13.003980 |  1.14 |
| p0.minicpm3-4b.FP16.inf.fp16.pt.llm_1024_prompt                                                     |  123.728982 |  108.500723 |  1.14 |
| - matmul                                                                                            |  123.728250 |  108.499991 |  1.14 |
| p0.bert-base-ner.default.inf.fp16.onnx.public62                                                     |   14.895570 |   13.069498 |  1.14 |
| - matmul                                                                                            |   14.895570 |   13.069498 |  1.14 |
| p0.bert-base-ner.inf.fp16.ov.mb1                                                                    |    0.796230 |    0.698810 |  1.14 |
| - matmul                                                                                            |    0.796230 |    0.698810 |  1.14 |
| p0.mistral-7b-instruct-v0.3.FP16.inf.fp16.pt.llm_1024_prompt                                        |  215.625320 |  189.260780 |  1.14 |
| - matmul                                                                                            |  215.624588 |  189.260048 |  1.14 |
| p0.mistral-7b-instruct-v0.2.FP16.inf.fp16.pt.llm_1024_prompt                                        |  215.583862 |  189.243696 |  1.14 |
| - matmul                                                                                            |  215.583130 |  189.242964 |  1.14 |
| p0.bert-base-cased.default.inf.fp16.onnx.public60                                                   |   14.888481 |   13.069498 |  1.14 |
| - matmul                                                                                            |   14.888481 |   13.069498 |  1.14 |
| p0.roberta-base.default.inf.fp16.onnx.public181                                                     |   14.884945 |   13.067730 |  1.14 |
| - matmul                                                                                            |   14.884945 |   13.067730 |  1.14 |
| p0.distilbert-base-uncased-sst-2.default.inf.fp16.onnx.public88                                     |    7.520358 |    6.604687 |  1.14 |
| - matmul                                                                                            |    7.520358 |    6.604687 |  1.14 |
| p0.roberta-base-mrpc.default.inf.fp16.onnx.public183                                                |   14.862012 |   13.064245 |  1.14 |
| - matmul                                                                                            |   14.862012 |   13.064245 |  1.14 |
| p0.falcon-7b-instruct.FP16.inf.fp16.pt.llm_1024_prompt                                              |  208.160940 |  183.503862 |  1.13 |
| - matmul                                                                                            |  208.160940 |  183.503862 |  1.13 |
| p1.sbert-base-mean-tokens.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx             |    4.374840 |    3.874740 |  1.13 |
| - inner_product                                                                                     |    4.049880 |    3.549780 |  1.14 |
| p1.bert-base-ner.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx                      |    4.393590 |    3.893490 |  1.13 |
| - inner_product                                                                                     |    4.068630 |    3.568530 |  1.14 |
| p1.roberta-base-mrpc.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx                  |    4.391000 |    3.892985 |  1.13 |
| - inner_product                                                                                     |    4.078520 |    3.580505 |  1.14 |
| p1.bert-base-cased.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx                    |    4.391505 |    3.894010 |  1.13 |
| - inner_product                                                                                     |    4.066545 |    3.569050 |  1.14 |
| p1.stable-diffusion-v1-4-text-encoder.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx |    3.918540 |    3.474780 |  1.13 |
| - inner_product                                                                                     |    3.343620 |    2.893620 |  1.16 |
| - matmul                                                                                            |    0.574920 |    0.581160 |  0.99 |
| p1.roberta-base.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx                       |    4.390465 |    3.893490 |  1.13 |
| - inner_product                                                                                     |    4.065505 |    3.568530 |  1.14 |
| p1.distilbert-base-uncased-sst-2.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx      |    2.212940 |    1.964975 |  1.13 |
| - inner_product                                                                                     |    2.053580 |    1.805615 |  1.14 |
| p1.bert-base-ner.ir.inf.fp16.ov.device-ubuntu22_xeon-gold-6338-atsm-framework-onnx                  |    4.368630 |    3.893490 |  1.12 |
| - inner_product                                                                                     |    4.037430 |    3.562290 |  1.13 |
| p0.phi-3-medium-4k-instruct.FP16.inf.fp16.pt.llm_1024_prompt                                        |  473.537588 |  422.904178 |  1.12 |
| - matmul                                                                                            |  473.537100 |  422.903690 |  1.12 |
| p0.wav2vec2-base.default.inf.fp16.onnx.public229                                                    |   19.425398 |   17.651210 |  1.10 |
| - matmul                                                                                            |   14.079196 |   12.305008 |  1.14 |
| p1.wav2vec2-base.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx                      |    5.992535 |    5.456610 |  1.10 |
| - inner_product                                                                                     |    3.604100 |    3.068175 |  1.17 |
| p0.meta-llama-3.1-8b-instruct.FP16_FP16.inf.fp16.pt.vllm_mb100                                      |  216.231136 |  197.740250 |  1.09 |
| - matmul                                                                                            |  216.231136 |  197.740250 |  1.09 |
| p0.deepseek-r1-distill-qwen-14b.FP16_FP16.inf.fp16.pt.vllm_mb18                                     |  488.396806 |  455.130104 |  1.07 |
| - matmul                                                                                            |  488.396806 |  455.130104 |  1.07 |
| p1.rnnt-joint.ir.inf.fp16.ov.device-xeon-platinum-8352Y-atsm-framework-onnx                         |    0.032810 |    0.030725 |  1.07 |
| - inner_product                                                                                     |    0.032810 |    0.030725 |  1.07 |

my custom perf testing (whole set of models w/ mode=P) failed again by timeout. I will run again only for models from the list with regression.
